test/relayclient/ContractInteractor.test.ts

```diff
@@ -28,4 +35,28 @@ contract('ContractInteractor', function () {
       await relayClient._init()
     })
   })
+
+  context('#broadcastTransaction()', function () {
+    let provider: ProfilingProvider
+    let contractInteractor: ContractInteractor
+    let sampleTransactionHash: PrefixedHexString
+    let sampleTransactionData: PrefixedHexString
+
+    before(async function () {
+      provider = new ProfilingProvider(web3.currentProvider as HttpProvider)
+      contractInteractor = new ContractInteractor(provider, configureGSN({}))
+      const nonce = await web3.eth.getTransactionCount('0xb473D6BE09D0d6a23e1832046dBE258cF6E8635B')
+      const transaction = new Transaction({ to: constants.ZERO_ADDRESS, gasLimit: '0x5208', nonce })
+      transaction.sign(Buffer.from('46e6ef4a356fa3fa3929bf4b59e6b3eb9d0521ea660fd2879c67bd501002ac2b', 'hex'))
+      sampleTransactionData = '0x' + transaction.serialize().toString('hex')
+      sampleTransactionHash = '0x' + transaction.hash(true).toString('hex')
+    })
+
+    it('should sent the transaction to the blockchain directly', async function () {
```

Comment:
you should test what the title says you test. I don't see you check the transaction was sent to the blockchain at all: e.g. getTransaction(sampleTransactionHash)
Currently, its a test for your provider.

Reply:
You see, web3.sendSignedTransaction would still pass the test, because the transaction does reach the network. 
This test is white box in terms of - I know tx hash is not calculated by the Contract Interactor, so I know the correct transaction reached the blockchain and if the tx hash matches.

I do test that 'broadcastTransaction' does **not** do anything else, and as I've said, negative tests are marginally useful (aka useless 😀)